app/Http/Controllers/ExportController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Signup;
+use Rogue\Services\ExportService;
+
+class ExportController extends Controller
+{
+    /**
+     * ExportService instance
+     *
+     * @var Rogue\Services\ExportService
+     */
+    protected $export;
+
+    /**
+     * Instantiate a new ExportController instance.
+     *
+     * @param Rogue\Services\Registrar $registrar
+     */
+    public function __construct(ExportService $export)
+    {
+        $this->middleware('auth');
+        $this->export = $export;
+    }
+
+    /**
+     * Download the export of signup details for the specified campaign.
+     *
+     * @param int $campaignId
+     */
+    public function show($campaignId)
+    {
+        // Run the query
+        $signups = Signup::whereNotNull('details')->where('campaign_id', $campaignId)->get();
+
+        // Compile the data
+        $final_results = $this->export->exportSignups($signups);
+
+        // Format as csv
```

Comment:
Not at all necessary for merge, but you might want to look into using [`league/csv`](http://csv.thephpleague.com) to create this CSV file since I think we might run into some edge-cases here when doing this ourselves (like if a user's name included a `,` for example).

Reply:
@DFurnes I feel like I should add this now rather than later so we don't end up in a case where the the file is all mucked up and then we have to hurry to add in that package to handle it better

Reply:
I think that's a good call! 👍